src/Chapter2.hs

```diff
@@ -740,7 +769,8 @@ the list with only those lists that contain a passed element.
 
 🕯 HINT: Use the 'elem' function to check whether an element belongs to a list
 -}
-contains = error "contains: Not implemented!"
+contains :: Int -> [[Int]] -> [[Int]]
+contains n xss = filter (\xs -> n `elem` xs) xss
```

Comment:
Now, after you've mastered eta-reduction, you can apply such a technique in here as well  (both on the function and lambda function) 🙂 
But your implementation is already great 💯

Reply:
Thanks for your feedback ! I indeed forgot about eta-reduction in this one, tbh I've done a bit of Haskell already, but I tend not to use too much eta-reduction (not that I don't like it, I just tend to forget about it).
Is eta-reduction considered a best practice in Haskell ? Does it offer any kind of optimization or is it purely to make code look more idiomatic ?

Reply:
I also don't usually advice to do too much of the eta-reduction. However, in many cases, it improves code readability and maintainability a lot! Especially it works well in partial functions applications inside other functions. As you see, lambda functions could be noisy sometimes. But, of course, it is harmless when you apply it to the standard functions. With custom and less known ones, it could be less beneficial as it could introduce only confusion in understanding.

So I can't give you the concrete answer with that. The best answer is to use it when it helps and it is better to be explicit if it doesn't harm 🙂